### ISSUE TITLE
kaplan meier survival estimates confidence interval (with error='greenwood' and conf.type='log')

### ISSUE THREAD

**Issue by lbraglia** [therneau/survival#43]
**TL;DR**

Given a question from a referee, is there a citation for how default confidence intervals (formulas for "log" conf.type with "greenwood" error) are calculated for Kaplan-Meier curves, since are "different"
from the log style ci proposed by Borgan and Liestol [1990] and presented in Klein and Moeschberger (KB, pag 105)

**Full version**

Regarding confidence interval for Kaplan-Meier curves, if one uses suggested/default settings ("greenwood" variance and "log" conf.type) the estimated confidence interval is, in pseudo-code (and to my understanding of the source code):

```
lower = exp( log(surv) - z * sqrt(greenwood) )          (1)
upper = exp( log(surv) + z * sqrt(greenwood) )
```

This is "different" from log trasformed way proposed by borgan and liestol [1990] and presented in KB (pag 105) which is

```
lower = surv^(1/theta)                                  (2)
upper = surv^theta
```

where

```
theta = exp( (z * sqrt(greenwood)) / log(surv)) 
```

Obviously "different" does not mean "worse", but I only would like to ask if (since a referee asked me) there's a citation/methodology for formulas (1) adopted.

The following example shows the differences in reproducing example 4.2 pag 105 from KB; the modified version is taken from [here] (https://github.com/lbraglia/survival_old/commit/9afa90a30503aecef69bb9abd49f1ee89e1cca56#diff-d06138dfd71c099d4186c1e138e24f60)
(where I ignored scale for a quick'n dirty check, and the starting source code is taken from current cran version)

```
> ci <- function(conf.type, error = 'greenwood'){
+     require(KMsurv)
+     data(bmt)
+     fit <- survfit(Surv(time = t2, event = d3) ~ group,
+                    conf.type = conf.type,
+                    error = error,
+                    data = bmt)
+     sfit <- summary(fit, time = 365)
+     ## table pag 106 klein moeschberger
+     rval <- with(sfit, cbind(lower, upper))
+     rownames(rval) <- c('ALL', 'AML low risk', 'AML high risk')
+     rval
+ }
> 
> ## ----------------
> ## standard version
> ## ----------------
> library(survival, lib.loc = '/usr/lib/R/library') ## std

> ci(conf.type = 'plain')         # identical
Loading required package: KMsurv
                  lower     upper
ALL           0.3900045 0.7083937
AML low risk  0.6668928 0.8886628
AML high risk 0.2361225 0.5194331

> ci(conf.type = 'arcsin')         # identical
                  lower     upper
ALL           0.3902099 0.7031949
AML low risk  0.6583810 0.8775309
AML high risk 0.2433091 0.5226060

> ci(conf.type = 'log')           # different
                  lower     upper
ALL           0.4110004 0.7338670
AML low risk  0.6744344 0.8969564
AML high risk 0.2596504 0.5496469
 
> ## --------------------------------
> ## changed log (Borgan and Liestol)
> ## --------------------------------
> detach('package:survival', unload = TRUE)
> library(survival, lib.loc = '~/R/x86_64-pc-linux-gnu-library/3.4')
> ci(conf.type = 'log')
                  lower     upper
ALL           0.3782973 0.6910998
AML low risk  0.6419883 0.8671777
AML high risk 0.2390992 0.5156879
```

References:

- Borgan Liestol, "A note on confidence intervals and bands for the survival curve based on transformations" - Scandinavian Journal of statistics 17 (1990):35-41
- Klein Moeschberger, "Survival analysis" (2nd edition)

**Comment by lbraglia** [comment]
Ok after a bit more checking, method (2) should be obtained out of the box with "log-log" `conf.type`. I was a bit misleaded by the KM naming :)
